writing-guide.md

```diff
@@ -0,0 +1,86 @@
+- **Almost always avoid humor (for English docs)**, especially sarcasm and pop culture references, as it doesn't translate well across cultures.
+- **Never assume a more advanced context than you have to.**
+- **In most cases, prefer links between sections of the docs over repeating the same content in multiple sections.** Some repetition in content is unavoidable and even essential for learning. However, too much repetition also makes the docs more difficult to maintain, because a change in the API will require changes in many places and it's easy to miss something. This is a difficult balance to strike.
+- **Specific is better than generic.** For example, a `<BlogPost>` component example is better than `<ComponentA>`.
+- **Relatable is better than obscure.** For example, a `<BlogPost>` component example is better than `<CurrencyExchangeSettings>`.
+- **Be emotionally relevant.** Explanations and examples that relate to something people have experience with and care about will always be more effective.
+
+### Grammar
+
+- **Avoid abbreviations** in writing and code examples (e.g. `attribute` is better than `attr`, `message` is better than `msg`), unless you are specifically referencing an abbreviation in an API (e.g. `$attrs`).
+- **When referencing a directly following example, use a colon (`:`) to end a sentence**, rather than a period (`.`).
+- **Use the oxford comma** (e.g. "a, b, and c" instead of "a, b and c"). ![Why the oxford comma is important](https://www.inkonhand.com/wp-content/uploads/2015/10/serialimage.jpg)
+- **When referencing the name of a project, prioritize the broader conventions of English over internal branding conventions of that project.** For example, "webpack" and "npm" both disregard conventions such as "always start a word at the beginning of a sentence with a capital letter", "project names always use Title Case", and "acronyms are always capitalized". Instead, always write "Webpack and NPM" to provide a more consistent experience in the docs and avoid sentences like "If you don't want to use Vue CLI, you can use webpack or Rollup directly by installing them via npm or Yarn".
+- **Use Title Case for headings** - at least for now, since it's what we use through the rest of the docs. There's research suggesting that sentence case (only first word of the heading starts with a capital) is actually superior for legibility and also reduces the cognitive overhead for documentation writers, since they don't have to try to remember whether to capitalize words like "and", "with", and "about".
+- **Always prefer simpler, plainer language over complex or jargony language.** For example:
+  - "you can use Vue with a script element" instead of "in order to initiate the usage of Vue, one possible option is to actually inject it via a script HTML element"
+  - "function that returns a function" instead of "higher order function"
```

Comment:
In this particular case, I'd argue that "higher-order function" is a well-known term that is expected to be used _for_ the correct target audience. "Function that returns a functions" is more verbose and at the same time [insufficient](https://en.wikipedia.org/wiki/Higher-order_function).

Reply:
I'm open to changing this example, but here's my thinking. While "higher-order function" and even "higher-order component" may be well-_known_ terms (i.e. many people - or at least those more experienced in the field or coming from a computer science background - have heard of them and even used them), I would argue they're not at all well-_understood_. If I did an offline poll in an advanced workshop/meetup/conference, I feel very confident that the majority of people would not be able to give a complete and accurate definition of either. It might even be _impossible_ to give an "accurate" definition, since they seem to have different definitions depending on where you look. 😄

And their technical definitions often differ even further from how they're used by working developers. For example, I've never heard someone refer to a function that just accepts a callback a "higher-order function", but it fits the technical criteria. So do many constructors, if you count functions that return an object that has methods on it (or all constructors, if you count getters and setters). The most widely accepted technical definition of a "higher-order function" is possibly so vague as to be useless.

By avoiding jargon like this altogether and getting right to what we mean in a particular use case with "function that returns a function", we're clearer _and_ more specific. At the cost of only two extra syllables, that seems like a deal! 😄

Do you disagree?

Reply:
If such a term is controversial, shouldn't we avoid it altogether and use another as an example?

Reply:
I don't think these terms are widely considered controversial - they're very widely used and I see very few people questioning their use.